src/time_travel/patchers/datetime_patcher.py

```diff
@@ -1,30 +1,246 @@
+    def today(cls):
+        """Return today's date."""
+        result = cls._now()
+        return date_to_fakedate(result)
+
+
+FakeDate.min = date_to_fakedate(_real_date.min)
+FakeDate.max = date_to_fakedate(_real_date.max)
+
+
+class FakeDatetime(with_metaclass(DatetimeSubclassMeta, 'datetime',
+                                  _real_datetime, FakeDate)):
+    """Mocked datetime.datetime class."""
+    
+    def __new__(cls, *args, **kwargs):
+        """Return a new mocked datetime object."""
+        return _real_datetime.__new__(cls, *args, **kwargs)
+
+    @classmethod
+    def now(cls, tz=None):
+        """Return a datetime object representing current time."""
+        now = cls._now()
+        if tz:
+            result = tz.fromutc(now.replace(tzinfo=tz)) +\
+                datetime.timedelta(hours=cls._tz_offset())
+        else:
+            result = now
+        return datetime_to_fakedatetime(result)
+
+    @classmethod
+    def today(cls):
+        """Return a datetime object representing current time."""
+        return cls.now(tz=None)
+
+    @classmethod
+    def utcnow(cls):
+        """Return a datetime object representing current time."""
+        result = cls._now()
+        return datetime_to_fakedatetime(result)
+
+
+FakeDatetime.min = datetime_to_fakedatetime(_real_datetime.min)
+FakeDatetime.max = datetime_to_fakedatetime(_real_datetime.max)
+
+
+def pickle_fake_date(datetime_):
+    """Pickle function for FakeDate."""
+    return FakeDate, (
+        datetime_.year,
+        datetime_.month,
+        datetime_.day,
+    )
+
+
+@staticmethod
+def pickle_fake_datetime(datetime_):
+    """Pickle function for FakeDatetime."""
+    return FakeDatetime, (
+        datetime_.year,
+        datetime_.month,
+        datetime_.day,
+        datetime_.hour,
+        datetime_.minute,
+        datetime_.second,
+        datetime_.microsecond,
+        datetime_.tzinfo,
+    )
 
 
 class DatetimePatcher(BasicPatcher):
     """Patcher of the datetime module.
     
     patching:
         - datetime.today
+        - datetime.now
+        - datetime.utcnow
+        - date.today
     """
     
-    def __init__(self, *args, **kwargs):
+    def __init__(self, patched_modules=None, **kwargs):
         """Create the patch."""
-        super(DatetimePatcher, self).__init__(*args, **kwargs)
+        super(DatetimePatcher, self).__init__(**kwargs)
+        
+        FakeDate._now = self._now
+        FakeDatetime._now = self._now
+        
+        if patched_modules is None:
+            self.patched_modules = []
+        elif isinstance(patched_modules, (list, tuple)):
+            self.patched_modules = patched_modules
+        else:
+            self.patched_modules = [patched_modules]
+        
+        self._undo_set = set()
         
-        self.datetime = mock.Mock(wraps=datetime_lib.datetime)
-        self.datetime.mock_add_spec(
-            ["today", "now", "utcnow", "fromtimestamp"])
+    def start(self):
```

Comment:
This method needs **a lot** more documentation. It's impossible to understand the thought process as it is now.